alpha/apps/kaltura/lib/kManifestRenderers.php

```diff
@@ -692,37 +692,63 @@ protected function getHeaders()
 		return array("Content-Type: application/x-mpegurl");
 	}
 	
+	protected function hasAudioFlavors() {
+	    foreach($this->flavors as $flavor) {
+	        if (isset($flavor['audioLanguage']))
+	            return true;
+	    }
+	    return false;
+	}
+	
 	/* (non-PHPdoc)
 	 * @see kManifestRenderer::getManifestFlavors()
 	 */
 	protected function getManifestFlavors()
 	{
+	    $audio = null;
+	    if ($this->hasAudioFlavors()) {
+	        $audio = ",AUDIO=\"audio\"";
+	    }
+	    
 		$flavorsArr = array();
 		foreach($this->flavors as $flavor)
 		{
-			$bitrate = (isset($flavor['bitrate']) ? $flavor['bitrate'] : 0) * 1024;
-			$codecs = "";
-			if ($bitrate && $bitrate <= 65536)
-				$codecs = ',CODECS="mp4a.40.2"';
-
-			// in case of Akamai HDN1.0 increase the reported bitrate due to mpeg2-ts overhead
-			if (strpos($flavor['url'], "index_0_av.m3u8"))
-				$bitrate += 40 * 1024;
-
-			$resolution = '';
-			if(isset($flavor['width']) && isset($flavor['height']))
-			{
-				$width = $flavor['width'];
-				$height = $flavor['height'];
-				if ($width && $height)
-					$resolution = ",RESOLUTION={$width}x{$height}";
-			}
-				
-			$content = "#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH={$bitrate}{$resolution}{$codecs}\n";
-			$content .= $flavor['url'];
-			$flavorsArr[] = $content;
+		    // Sperate audio flavors from video flavors
+		    $language = $flavor['audioLanguage'];
+		    if (isset($language)) {
+		        $languageName = constant('LanguageKey::' . strtoupper($language));
```

Comment:
Please use defined (php function) to verify that the constant exists to avoid php errors.